Cards/Movement State/MoveState_Queued.gd

```diff
@@ -23,10 +25,18 @@ func on_state_enter() -> void:
 # @Override
 func on_state_process(delta: float) -> void:
 	# Ease position with an offset
-	var offset_desired_position: Vector2 = _state.desired_position
+	var offset_desired_position: Vector2 = _state.card.get_global_mouse_position()
 	offset_desired_position.y += QUEUE_OFFSET
+	offset_desired_position -= _state.card.get_parent().position
+	offset_desired_position += OFFSET_FROM_MOUSE
+	
+	var is_in_play_area : bool = CardManager.card_container.is_queued_card_in_play_area()
+	var is_insta_cast : bool = _state.card.card_cast_type != Enums.CardCastType.INSTA_CAST
 	
-	_state.card.position = _state.card.position.lerp(offset_desired_position, delta * MOVE_SPEED)
+	if(is_in_play_area && is_insta_cast):
+		_state.card.position = _state.card.position.lerp(TARGET_CARD_POS, delta * MOVE_SPEED)
+	else :
+		_state.card.position = _state.card.position.lerp(offset_desired_position, delta * MOVE_SPEED)
```

Comment:
Maybe we should introduce a new `TARGETING` card state. So the idea is that when you place a card in the play area, if it's `SINGLE_TARGET`, then it enters the `TARGETING` state. Right now, the card will move to the `TARGET_CARD_POSITION` immediately when it enters the play area, which is a bit confusing to me.

Reply:
Actually playing with it a bit more, I think it's fine in the current state. I'd be curious to hear other people's thoughts though

Reply:
I think it will feel better if we add some more effects to make it less confusing, I think it's better that it doesn't require an extra click

Reply:
Yes I think effects will help a lot. I tested in StS as well and it works pretty much exactly like you have it here